airbyte-server/src/main/java/io/airbyte/server/cache/DefaultSpecCache.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.server.cache;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import java.util.Optional;
+
+public class DefaultSpecCache implements SpecCache {
+
+  private final Cache<String, ConnectorSpecification> cache;
+
+  public DefaultSpecCache() {
+    cache = CacheBuilder.newBuilder().build();
+  }
+
+  @Override
+  public Optional<ConnectorSpecification> get(String imageName) {
+    return Optional.ofNullable(cache.getIfPresent(imageName));
+  }
+
+  @Override
+  public void put(String imageName, ConnectorSpecification spec) {
```

Comment:
opted not to use the LoadingCache that guava has (where it does fetch-if-absent behavior). mostly, i didn't like the error handling. 

this whole iface exists though because i want to be able to change my impl easily if i decide i messed up here.